### ISSUE TITLE
Waydroid Halium 11 setup is non-trivial

### ISSUE THREAD

**Issue by Idcrafter** [HelloVolla/ubuntu-touch-beta-tests#132]
### Is this a new report?

Yes

### Device model

Volla Phone 22 (2022)

### OS version

16.04 (2022-09-02)

### Describe the bug

waydroid init does allways give a error that https://github.com/waydroid/OTA/blob/master/vendor/waydroid_arm64/HALIUM_11.json
doesn't exist and then does nothing, i checked and there is no file but using a custom image isn't workin either because for onces that i know work would use up 4,7GB which is a small bit more as the free space in /usr and other don't work with the waydroid log saying Clipboardmanager hasn't started and it just don't work 

### Steps to reproduce the bug

1. install waydroid via the waydroid helper application form the openstore or via apt 
2. run as root waydroid init or use as root waydroid init -f
3. then start waydroid or the waydroid session 

### Expected behaviour

it will show the application start screen forever or crash and show in the log that it has exited

**Comment by JamiKettunen** [comment]
Related: https://gitlab.com/ubports/development/core/hybris-support/lxc-android-config/-/issues/36

To use custom Android 11 images currently you basically bind-mount a `userdata` directory on `/usr/share/waydroid-extra` after creating it similar to what's done to `/var/lib/waydroid` already automatically:
```
mkdir -p {/userdata/system-data,}/usr/share/waydroid-extra
chmod 777 /userdata/system-data/usr/share/waydroid-extra
mount -o bind /userdata/system-data/usr/share/waydroid-extra /usr/share/waydroid-extra
```
Afterwards you can `mkdir -p /usr/share/waydroid-extra/images` and use that as per https://docs.waydro.id/faq/using-custom-waydroid-images. To make the bind-mount persistent you can `echo '/usr/share/waydroid-extra               auto                    persistent  none        none' >> /etc/system-image/writable-paths`

Additionally I've done `sed 's/aidl2/aidl3/g' -i /etc/gbinder.d/anbox.conf` but that might be unnecessary.

Additionally before Waydroid 1.3.1 (UBports has 1.2.1 in repos) you also need to tweak `/var/lib/waydroid/waydroid_base.prop` after `waydroid init -f` and set:
```
ro.hardware.gralloc=android
ro.hardware.egl=meow
```
Then `sudo restart waydroid-container`.

**Comment by JamiKettunen** [comment]
@Idcrafter If you don't mind could you link which Android 11 images you used? I'm sure that'll come in handy for users wanting to try this early :)

The images can also be shrunk further after download with `resize2fs -M <path to system or vendor>.img` so they don't have to take up so much space either.

**Comment by Idcrafter** [comment]
the link is now down but it was on some kind of blissoss or blissoss-dev source forge, but there is now only X86 but i think the images might still be somewhere else but thanks for all the help and if i find any other image again will i then update this

**Comment by JamiKettunen** [comment]
Let's keep this open until official Android 11 image installation is possible with just `waydroid init` and Waydroid 1.3.1+ is available on UT which is required to avoid having to manually edit `waydroid_base.prop` to get working graphics output :)

**Comment by pnguyen9** [comment]
Is this the reason why Waydroid isn't booting at all on my Volla Phone 22 when installed with Waydroid Helper? Or should I open a separate issue?

**Comment by JamiKettunen** [comment]
@pnguyen9 Most likely yes, from what I can tell [it just tries `waydroid init`](https://github.com/Aarontheissueguy/WaydroidHelper/blob/89568bc/src/installer.py#L41-L51) which should result in this exact same error about missing https://github.com/waydroid/OTA/blob/master/vendor/waydroid_arm64/HALIUM_11.json seen here, but I guess that isn't shown in the app UI for some reason...

https://open-store.io/app/waydroidhelper.aaronhafer doesn't mention Volla Phone 22 as a supported device either, though in my opinion a better wording would be "verified working devices".

**Comment by JamiKettunen** [comment]
Now both official Android 11 system as well as HALIUM11 vendor images for Waydroid are available; making the setup process trivial (i.e. click of a button in [Waydroid Helper](https://open-store.io/app/waydroidhelper.aaronhafer) requires upgrading Waydroid related packages such as [`libgbinder`](https://gitlab.com/ubports/development/core/packaging/libgbinder) (>=1.1.20), [`python-gbinder`](https://gitlab.com/ubports/development/core/packaging/gbinder-python) (>=1.1.0) and [`waydroid`](https://gitlab.com/ubports/development/core/packaging/waydroid) (>=1.3.0) which is currently blocked by OTA-24 merge freeze.

For now though, running Waydroid involves the following steps:
```sh
sudo su     # to get a root shell and avoid having to type "sudo" before every command below

mount -o remount,rw /
apt update && apt install -y waydroid && waydroid init

# https://github.com/waydroid/waydroid/commit/a3d741d ("Detect HIDL gralloc4")
# makes the following unneeded, Waydroid 1.3.1+
sed -i /var/lib/waydroid/waydroid_base.prop \
    -e 's/^ro.hardware.gralloc=.*/ro.hardware.gralloc=android/' \
    -e 's/^ro.hardware.egl=.*/ro.hardware.egl=meow/'

# https://github.com/waydroid/waydroid/commit/5eb95c3 ("Set aidl version based on the android version")
# makes the following unneeded, Waydroid 1.3.0+
sed '/aidl/s/2/3/' -i /etc/gbinder.d/anbox.conf

mount -o remount,ro /
stop waydroid-container     # only if you already started it before these fixes
start waydroid-container
exit     # back to non-root "phablet" user

waydroid show-full-ui     # clicking on the "Waydroid" launcher icon at this point would do the same :)
```
**NOTE:** At least one user has reported additionally needing a full `reboot` before Waydroid started launching!

**Comment by koka440** [comment]
thanks a lot for finally bringing waydroid to my Volla 22. The vanilla image update from the day before yesterday seems to be corrupted. installing the GAPPS version using 'waydroid init -s GAPPS' though does work.

**Comment by JamiKettunen** [comment]
> thanks a lot for finally bringing waydroid to my Volla 22. The vanilla image update from the day before yesterday seems to be corrupted. installing the GAPPS version using 'waydroid init -s GAPPS' though does work.

@koka440 Ah, unfortunate timing for you to try Waydroid VANILLA images, it seems there was an issue with 170 byte images getting uploaded for various architectures which was fixed, but latest builds for `arm64` failed for now. I suppose keep an eye on https://sourceforge.net/projects/waydroid/files/images/system/lineage/waydroid_arm64/ (or https://github.com/waydroid/OTA/commits/master/system/lineage/waydroid_arm64) if you're still interested in running `VANILLA` images instead of with `GAPPS`

**Comment by JamiKettunen** [comment]
There's further progress on this issue and it should be solved soon once and for all; before installing Waydroid through command line or [Waydroid Helper](https://open-store.io/app/waydroidhelper.aaronhafer) the following can simply be ran to start using the upgraded Waydroid packages I've been working on:
```sh
sudo ubports-qa install PR_gbinder-python_2
sudo ubports-qa install PR_waydroid_5
```
No other manual steps for editing files etc. should be needed if the initial setup is done like this. For more details see https://ubports.atlassian.net/browse/YUMI-11

**NOTE:** The above commands only work on Development update channel, otherwise please wait until the updates are merged to get them elsewhere!

**Comment by JamiKettunen** [comment]
The related pull requests are now merged and updated `python3-gbinder` and `waydroid` packages are available from the repository even on 16.04 OTA-24, which means [Waydroid Helper](https://open-store.io/app/waydroidhelper.aaronhafer) should just work now on Volla Phone 22 for the first time :)
